### ISSUE TITLE
Add `libwebp` for Decoding WebP raster tiles

### ISSUE THREAD

**Issue by roblabs** [maplibre/maplibre-native#58]
> *[WebP](https://developers.google.com/speed/webp) is an open source image format that provides smaller file sizes and lossy & lossless formats.*

* See the branch [roblabs/maplibre-gl-native:webp-ios](https://github.com/roblabs/maplibre-gl-native/blob/webp-ios/platform/ios/platform/ios/vendor/webp/README-WebP.md) for details on how to bring WebP into MapLibre.  The decision was made to document how to do this, rather than a formal PR at this time.

* a basic style for adding a WebP raster layer for Joshua Tree National Park.

```json
{
  "version": 8,
  "name": "Joshua Tree National Park...in WebP",
  "center": [-115.746, 33.866],
  "zoom": 7,
  "sources": {
    "jotr": {
        "tiles": [ "asset://joshuatree/{z}/{x}/{y}.webp" ],
        "type": "raster"
    }
  },
  "layers": [
    {
      "id": "background",
      "type": "background",
      "paint": {
        "background-color": "#ddeeff"
      }
    },
    {
        "id": "jotr",
        "type": "raster",
        "source": "jotr",
        "layout": {},
        "paint": {}
    }
  ]
}
```

This work is donated by [ePi Rational, Inc.](https://roblabs.com/webp/) based on work initially done by [mapbox/mapbox-gl-native](https://github.com/mapbox/mapbox-gl-native)

And updated over several versions of the iOS SDK.

* [WebP for Mapbox SDK for iOS v3.6.0](https://github.com/roblabs/mapbox-gl-native/blob/546374c0149d5ce6bdaa2aba595cc5e81b1dbcbf/README.md) in July 2017
* [WebP for Mapbox SDK for iOS v5.1.1](https://github.com/roblabs/mapbox-gl-native/commits/Features/ios-v5.1.1-webp) in July 2019

---

*Example of WebP rasters in MapLibre for Joshua Tree National Park*
<img src="https://user-images.githubusercontent.com/118112/112783126-7492a000-9003-11eb-9bc4-8a5486b949c1.png" width=50%>

**Comment by roblabs** [comment]
### *Test Results of Archive & upload to App Store Connect (TestFlight) of the MapLibre test app `iosapp` with MapLibre + WebP*

**Results Summary** — the WebP SDK adds approximately 0.4 MB to the install size for iOS & iPadOS variants.

Key

* 🗺 - Size of `iosapp` with MapLibre SDK only
* 🗺+🕸 - Size with MapLibre SDK + WebP SDK
* The *Universal Device Type* has data for a basic MapKit app based on Swift UI.  This is intended to demonstrate the smallest possible app that can render a MapKit based map.

Methodology

* Upload a build to App Store Connect with *app thinning*.
* App Store Connect reports App Store File Sizes in KB.
* app thinning "*creates variants of your app for different devices and operating system versions. You can view the size of the variants directly in App Store Connect.*"<sup>1</sup>
* Approved apps for the App Store are "*encrypted with DRM and recompressed. The added encryption and DRM affects the ability to compress your binary, and as a result you may see a larger App Store file size for your binary than the binary you uploaded in App Store Connect.*"<sup>1</sup>


* <sup>1</sup> - Definitions such as *Download* and *Install* size - https://help.apple.com/app-store-connect/#/dev3b56ce97c?sub=dev7379ba2ed


DEVICE TYPE	 | DOWNLOAD SIZE | INSTALL SIZE
:-- | :-- | :--
| | *This is the compressed size of the app downloaded over the air.<sup>1</sup>* |  *This is the amount of disk space the app will take up on the customer's device.  The install size also appears on your app’s product page in the Information section.<sup>1</sup>*
Universal	|  `1706` KB MapKit<br> `11742` KB  🗺<br>**`12443`** KB 🗺+🕸<br>---<br>🕸 adds an additional 701 KB	|  `1924` KB MapKit<br> `20146` KB  🗺<br>**`20904`** KB 🗺+🕸<br>---<br>🕸  adds an additional 758 KB
iPad (4th generation)	|  `6085` KB  🗺<br>**`6411`** KB 🗺+🕸<br>---<br>🕸 adds an additional 326 KB	|  `13871` KB  🗺<br>**`14219`** KB 🗺+🕸<br>---<br>🕸 adds an additional 348 KB
iPad mini 2	|  `6221` KB  🗺<br>**`6598`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14105` KB  🗺<br>**`14502`** KB 🗺+🕸<br>---<br>🕸 adds an additional 397 KB
iPad mini 3	|  `6221` KB  🗺<br>**`6598`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14105` KB  🗺<br>**`14502`** KB 🗺+🕸<br>---<br>🕸 adds an additional 397 KB
iPad Pro (10.5‑inch)	|  `6221` KB  🗺<br>**`6598`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14105` KB  🗺<br>**`14502`** KB 🗺+🕸<br>---<br>🕸 adds an additional 397 KB
iPad (6th generation)	|  `6221` KB  🗺<br>**`6598`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14105` KB  🗺<br>**`14502`** KB 🗺+🕸<br>---<br>🕸 adds an additional 397 KB
iPad Pro (11‑inch)	|  `6220` KB  🗺<br>**`6597`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14100` KB  🗺<br>**`14498`** KB 🗺+🕸<br>---<br>🕸 adds an additional 398 KB
iPad Pro (12.9‑inch) (4th generation)	|  `2834` KB  🗺<br>**`3034`** KB 🗺+🕸<br>---<br>🕸 adds an additional 200 KB	|  `14007` KB  🗺<br>**`14396`** KB 🗺+🕸<br>---<br>🕸 adds an additional 389 KB
iPad mini (5th generation)	|  `6220` KB  🗺<br>**`6597`** KB 🗺+🕸<br>---<br>🕸 adds an additional 377 KB	|  `14100` KB  🗺<br>**`14498`** KB 🗺+🕸<br>---<br>🕸 adds an additional 398 KB
iPhone 5S	|  `6216` KB  🗺<br>**`6594`** KB 🗺+🕸<br>---<br>🕸 adds an additional 378 KB	|  `14096` KB  🗺<br>**`14494`** KB 🗺+🕸<br>---<br>🕸 adds an additional 398 KB
iPhone 7 Plus	|  `6225` KB  🗺<br>**`6603`** KB 🗺+🕸<br>---<br>🕸 adds an additional 378 KB	|  `14105` KB  🗺<br>**`14502`** KB 🗺+🕸<br>---<br>🕸 adds an additional 397 KB
iPhone 11	|  `2835` KB  🗺<br>**`3034`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14007` KB  🗺<br>**`14396`** KB 🗺+🕸<br>---<br>🕸 adds an additional 389 KB
iPhone 11 Pro	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPhone 11 Pro Max	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPhone SE (2nd generation)	|  `2830` KB  🗺<br>**`3029`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14003` KB  🗺<br>**`14392`** KB 🗺+🕸<br>---<br>🕸 adds an additional 389 KB
iPhone 12 mini	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPhone 12	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPhone 12 Pro	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPhone 12 Pro Max	|  `2839` KB  🗺<br>**`3038`** KB 🗺+🕸<br>---<br>🕸 adds an additional 199 KB	|  `14011` KB  🗺<br>**`14401`** KB 🗺+🕸<br>---<br>🕸 adds an additional 390 KB
iPod touch (6th generation)	|  `6216` KB  🗺<br>**`6594`** KB 🗺+🕸<br>---<br>🕸 adds an additional 378 KB	|  `14096` KB  🗺<br>**`14494`** KB 🗺+🕸<br>---<br>🕸 adds an additional 398 KB

**Comment by louwers** [comment]
According to [this comment](https://github.com/maplibre/maplibre-native/pull/1314) from @mwilsnd this is supported on iOS by now. Let me know if it's not.

**Comment by roblabs** [comment]
Closed from #1314
